src/App.jsx

```diff
@@ -3,46 +3,62 @@ import React, { useState, useEffect } from 'react';
 import AddTodoForm from './AddTodoForm';
 import TodoList from './TodoList';
 
-function useSemiPersistentState() {
-  const [todoList, setTodoList] = useState(
-    JSON.parse(localStorage.getItem('savedTodoList')) || []
-  );
-
-  useEffect(() => {
-    localStorage.setItem('savedTodoList', JSON.stringify(todoList));
-  }, [todoList]);
-
-  return [todoList, setTodoList];
-}
-
 function App() {
-  const [todoList, setTodoList] = useSemiPersistentState();
+  const [todoList, setTodoList] = useState(JSON.parse(localStorage.getItem('todoList')) ?? []);
+  const [isLoading, setIsLoading] = useState(true);
 
+  // Load initial data and save to localStorage
+  useEffect(() => {
+    const fetchData = async () => {
+      try {
+        const url = `https://api.airtable.com/v0/${import.meta.env.VITE_AIRTABLE_BASE_ID}/${import.meta.env.VITE_TABLE_NAME}`;
+        const options = {
+          method: 'GET',
+          headers: {
+            Authorization: `Bearer ${import.meta.env.VITE_AIRTABLE_API_TOKEN}`
+          }
+        };
+        const response = await fetch(url, options);
+        if (!response.ok) {
+          throw new Error('Error: ${response.status}');
+        }
+        const data = await response.json();
+        const todos = data.records.map(record => ({
+          id: record.id,
+          title: record.fields.title,
+          createTime: record.createdTime
+        }));
+        //console.log('Todos:', todos); 
+        setTodoList(todos);
```

Comment:
I would check the notes from the previous PR. You can run into issue(although rarely) if you directly pass an array into a useState setter function. The setter function expects a copy of the object which is important to always trigger a re-render.

Reply:
Create a new Function to update the todoList state `  const updateTodoList = (newTodoList) => {
    setTodoList([...newTodoList]);
  };`  and from the same updated the add and remove